sdk/keyvault/keyvault-admin/src/lro/backup/operation.ts

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortSignalLike } from "@azure/abort-controller";
+import { PollOperationState, PollOperation } from "@azure/core-lro";
+import { KeyVaultClient } from "../../generated/keyVaultClient";
+import {
+  KeyVaultClientFullBackupOptionalParams,
+  KeyVaultClientFullBackupResponse,
+  KeyVaultClientFullBackupStatusResponse
+} from "../../generated/models";
+import { createSpan, setParentSpan } from "../../tracing";
+import { BeginBackupOptions } from "../../backupClientModels";
+
+/**
+ * An interface representing the publicly available properties of the state of a backup Key Vault's poll operation.
+ */
+export interface BackupOperationState extends PollOperationState<string> {
+  /**
+   * Identifier for the full backup operation.
+   */
+  jobId?: string;
+  /**
+   * Status of the backup operation.
+   */
+  status?: string;
+  /**
+   * The status details of backup operation.
+   */
+  statusDetails?: string;
+  /**
+   * The start time of the backup operation in UTC
+   */
+  startTime?: Date;
+  /**
+   * The end time of the backup operation in UTC
+   */
+  endTime?: Date;
+}
+
+/**
+ * An internal interface representing the state of a backup Key Vault's poll operation.
+ * @internal
+ */
+export interface BackupPollOperationState extends PollOperationState<string> {
+  /**
+   * Options for the core-http requests.
+   */
+  requestOptions?: BeginBackupOptions;
+  /**
+   * An interface representing the internal KeyVaultClient.
+   */
+  client?: KeyVaultClient;
+  /**
+   * The base URL to the vault.
+   */
+  vaultUrl?: string;
```

Comment:
The issue here is that if we make this required, the BackupOperationState type can't be used since it's not a subset of this type :/

I think I could get away with it if I don't have it in the state but only as parameters. I'll try to do that.

Reply:
I did fix it! Please review again.